Pages/Index.cshtml.cs

```diff
@@ -13,40 +13,69 @@ namespace NewYork_CIty_School_Data_With_Crime_Rate_History.Pages
 {
     public class IndexModel : PageModel
     {
+        public string GetData(string endpoint)
+        {
+            string downloadedData = "";
+            using (WebClient webClient = new WebClient())
+            {
+                downloadedData = webClient.DownloadString(endpoint);
+            }
+            return downloadedData;
+        }
         public JsonResult OnGet()
         {
 
             List<MergedData> mergedData = new List<MergedData>();
 
             using (WebClient webClient = new WebClient())
             {
-                string schoolJsonString = webClient.DownloadString("https://data.cityofnewyork.us/resource/23z9-6uk9.json");
+                string schoolJsonString = GetData("https://data.cityofnewyork.us/resource/23z9-6uk9.json");
+                //               string schoolJsonString = webClient.DownloadString("https://data.cityofnewyork.us/resource/23z9-6uk9.json");
                 SchoolData[] schoolData = SchoolData.FromJson(schoolJsonString);
 
-                string crimeJsonString = webClient.DownloadString("https://data.cityofnewyork.us/resource/kwvk-z7i9.json");
+                string crimeJsonString = GetData("https://data.cityofnewyork.us/resource/kwvk-z7i9.json");
+                //                string crimeJsonString = webClient.DownloadString("https://data.cityofnewyork.us/resource/kwvk-z7i9.json");
                 CrimeData[] crimeData = CrimeData.FromJson(crimeJsonString);
-                              
 
-                // iterate over the specimens, to find which ones like water.
-                foreach (SchoolData schoolDatum in schoolData)
+                var schoolQuery = from SchoolData in schoolData
+                                   join CrimeData in crimeData
+                                   on SchoolData.Dbn equals CrimeData.Dbn
+                                   select new
+                                   {
+                                       Dbn = SchoolData.Dbn,
+                                       Address = CrimeData.Address
+                                   };
+                foreach (var item in schoolQuery)
                 {
-                    // find the matching plant record for this specimen.
-                    foreach (CrimeData crimeDatum in crimeData)
+                    mergedData.Add(new MergedData
                     {
-                        if (schoolDatum.Dbn == crimeDatum.Dbn)
-                        {
-                            var x = new MergedData();
-                            x.Dbn = schoolDatum.Dbn;
-                            x.Address = crimeDatum.Address;
+                        Dbn = item.Dbn,
+                        Address = item.Address
+                    });
+                }
```

Comment:
I am doing a Linq query Join instead of the Nested For loop. Nested Loops are generally inefficient. Another way, probably much more efficient way would be to implement a Dictionary, but I am not sure how it would work with string as Key.

Reply:
Strings can generate hashcodes... would be fine.